src/GoblinForm.js

```diff
@@ -1,29 +1,40 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
+export default function GoblinForm({ submitGoblin, onGoblinChange }) {
+  const [goblin, setGoblin] = useState({});
+
+  useEffect(() => {
+    onGoblinChange(goblin);
+  }, [goblin, onGoblinChange]);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    submitGoblin(goblin);
+  };
 
-export default function GoblinForm(props) {
   return (
     <div className='goblin-form-container quarter'>
       {/* on submit, call the submitGoblin function, passed in as a prop.
       Note that you don't need to define an anonymous function--you can just name the submitGoblin prop here and it will work.
       Take a minute to try and puzzle out why that is: how is this function different from other functions, 
       where you do need to create an anonymous function in the `onClick` or `onSubmit` spot? */}
-      <form className='goblin-form'>
+      <form onSubmit={handleSubmit} className='goblin-form'>
         <label>
             Name
           {/* onChange, use the prop setGoblinFormName to set the parent state */}
-          <input required value={props.goblinFormName} />
+          <input required onChange={e => setGoblin({ ...goblin, name: e.target.value })} value={goblin.name || ''} />
           {/* note that we're controlling the input's value from parent state */}
         </label>
         <label>
             HP
           {/* onChange, use the prop setGoblinFormHP to set the parent state */}
-          <input required type="number" value={props.goblinFormHP} />
+          <input required type="number" onChange={e => setGoblin({ ...goblin, hp: e.target.value })} value={goblin.hp || ''} />
           {/* note that we're controlling the input's value from parent state */}
         </label>
         <label>
             Color
           {/* onChange, use the prop setGoblinFormColor to set the parent state */}
-          <select required value={props.goblinFormColor}>
+          <select required onChange={e => setGoblin({ ...goblin, color: e.target.value })} value={goblin.color || ''}>
```

Comment:
YES, love how you spread in the goblin and just change the updated goblin. very cool solution